### PR TITLE
MnemonicSwift added to the project

### PR DESCRIPTION
**Pull request by LukasKorba**
Issue #155, MnemonicSwift SPM added + implementation of the MnemonicSeedPhraseHandling protocol. The next step is to implement KeyStoring protocol (#157) in order to start to put things together and use MnemonicSeedPhaseProvider.

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [x] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

**Comment by pacu** [comment]
I think that given TCA's functional nature the [interactor](https://github.com/zcash-hackworks/MnemonicSwift/blob/master/MnemonicSwift/MnemonicInteractor.swift) will be a better fit than using the MnemonicSwiftHandling protocol directly. 

What are your thoughts around that?

**Comment by LukasKorba** [comment]
> I think that given TCA's functional nature the [interactor](https://github.com/zcash-hackworks/MnemonicSwift/blob/master/MnemonicSwift/MnemonicInteractor.swift) will be a better fit than using the MnemonicSwiftHandling protocol directly.
> 
> What are your thoughts around that?

My mistake I overlooked this wrapper, agreed, this is useful and I'll refactor it.

File: secant/Util/MnemonicSeedPhraseProvider.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+//  MnemonicSeedPhraseProvider.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 03/09/2022.
+//
+
+import Foundation
+import MnemonicSwift
+
+class MnemonicSeedPhaseProvider: MnemonicSeedPhraseHandling {
```
Comment (adamstener) [inline secant/Util/MnemonicSeedPhraseProvider.swift:11]: Have you considered replacing the MnemonicSeedPhraseHandling protocol with the usage of the newer simple struct in MnemonicSwift the `MnemonicInteractor`?

Comment (adamstener) [reply secant/Util/MnemonicSeedPhraseProvider.swift:11]: (just noticed this was previously commented on, sorry for the duplication)

**Comment by LukasKorba** [comment]
After discussions with Adam and Pacu I made a decision to rewrite the MnemonicSeedPhraseProvider so it's:
- TCA syntax
- testable
- simplified for the use on the wallet side: to me the key element is the hardcoded strength 256, the wallet can use only randomMnemonic() and receive exactly 24 words.

I removed previous dependency and protocol MnemonicSeedPhraseHandling and cleaned up the code.

File: secant/Util/MnemonicSeedPhraseProvider.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            try Mnemonic.validate(mnemonic: mnemonic)
+        }
+    )
+    
+    static let mock = MnemonicSeedPhraseProvider(
+        randomMnemonic: {
+            """
+            still champion voice habit trend flight \
+            survey between bitter process artefact blind \
+            carbon truly provide dizzy crush flush  \
```
Comment (pacu) [inline secant/Util/MnemonicSeedPhraseProvider.swift:50]: I think this line has a double space and will (and probably should) break separating the string by components using space as a separator

File: secant/Util/MnemonicSeedPhraseProvider.swift [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            breeze blouse charge solid fish spread
+            """
+        },
+        randomMnemonicWords: {
+            let mnemonic = """
+                still champion voice habit trend flight \
+                survey between bitter process artefact blind \
+                carbon truly provide dizzy crush flush  \
```
Comment (pacu) [inline secant/Util/MnemonicSeedPhraseProvider.swift:58]: same as above

File: secant/Util/MnemonicSeedPhraseProvider.swift [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                base64Encoded: "9VDVOZZZOWWHpZtq1Ebridp3Qeux5C+HwiRR0g7Oi7HgnMs8Gfln83+/Q1NnvClcaSwM4ADFL1uZHxypEWlWXg=="
+            )!// swiftlint:disable:this force_unwrapping
+            
+            return [UInt8](seedString)
+        },
+        asWords: { mnemonic in
+            let mnemonic = """
+                still champion voice habit trend flight \
+                survey between bitter process artefact blind \
+                carbon truly provide dizzy crush flush  \
```
Comment (pacu) [inline secant/Util/MnemonicSeedPhraseProvider.swift:75]: same as above

**Review by pacu — CHANGES_REQUESTED** [review]

**Comment by pacu** [comment]
optional reviewer DevSecOps @defuse

File: secant/Util/MnemonicSeedPhraseProvider.swift (hunk 2 above)
Comment (LukasKorba) [reply secant/Util/MnemonicSeedPhraseProvider.swift]: fixed

File: secant/Util/MnemonicSeedPhraseProvider.swift (hunk 3 above)
Comment (LukasKorba) [reply secant/Util/MnemonicSeedPhraseProvider.swift]: fixed

File: secant/Util/MnemonicSeedPhraseProvider.swift (hunk 4 above)
Comment (LukasKorba) [reply secant/Util/MnemonicSeedPhraseProvider.swift]: fixed

**Review by pacu — APPROVED** [review]

**Review by dh-ecc — APPROVED** [review]
Looks great 👍.

For reference incase it comes up where we need to get words that might end up with >1 space in between them or something like that, `Foundation` has `https://developer.apple.com/documentation/foundation/nsstring/1416774-enumeratesubstrings` which you can use the `.byWords` option with to enumerate through them 😄.

**Review by adamstener — APPROVED** [review]

**Pull request merged**